api/cmd/portainer/main.go

```diff
@@ -212,12 +214,27 @@ func main() {
 		}
 	}
 
-	if *flags.AdminPassword != "" {
-		log.Printf("Creating admin user with password hash %s", *flags.AdminPassword)
+	AdminPasswordHash := ""
+	if *flags.AdminPasswordFile != "" {
+		content, err := ioutil.ReadFile(*flags.AdminPasswordFile)
```

Comment:
I'd rather see this in the `file` package rather to have an import to `ioutil` in this file.

Reply:
The `file` package has a `Service Struct` that operates on files that are relative to the `/data` path, and in case of passing secrets the files are provided with the full path which is usually not in the `/data`. If I move this to the file I would then create a global function on getting the string from a file only... Let me know if that is what you had in mind?

Reply:
Yeap, include it as a global generic function that returns a string from a file.